ntbs-service/Jobs/DataQualityAlertsJob.cs

```diff
@@ -22,75 +23,238 @@ public class DataQualityAlertsJob
             _alertService = alertService;
             _dataQualityRepository = dataQualityRepository;
         }
-
-        public async Task Run(IJobCancellationToken token)
+        
+        private async Task CreateDraftAlertsInBulk()
         {
-            Log.Information($"Starting data quality alerts job");
-
-            var notificationsForDraftAlerts =
-                await _dataQualityRepository.GetNotificationsEligibleForDataQualityDraftAlerts();
-            foreach (var notification in notificationsForDraftAlerts)
+            var notificationsForAlertsCount = await _dataQualityRepository.GetNotificationsEligibleForDataQualityDraftAlertsCount();
+            var x = 0;
+            while (x < notificationsForAlertsCount)
             {
-                var alert = new DataQualityDraftAlert {NotificationId = notification.NotificationId};
-                await _alertService.AddUniqueAlertAsync(alert);
+                var alertsToAdd = new List<Alert>();
+                var notificationsForAlerts =
+                    await _dataQualityRepository.GetMultipleNotificationsEligibleForDataQualityDraftAlerts(CountPerBatch, x);
+                foreach (var notification in notificationsForAlerts)
+                {
+                    
+                    var alert = new DataQualityDraftAlert {NotificationId = notification.NotificationId};
+                    var populatedAlert = await _alertService.CheckIfAlertExistsAndPopulateIfItDoesNot(alert, notification);
```

Comment:
Also I found `CheckIfAlertExistsAndPopulateIfItDoesNot` as a little bit of an odd "logical" unit and it got me thinking...
Perhaps we should combine it with the alert creation and simply make the alertService reponsible for giving us the appropriate alert - if it makes sense to do so. I'm thinking something like:
```
var alert = _alertService.CreateNew(DataQualityDraftAlert, notification);
```
and the implemenation would take care of checking if its unique:
```
public Alert CreateNew(Type alertType, notification)
{
  // .. do the check for uniqueness and return null if needed
  Alert alert = Activator.Createnstance(alertType);
  alert.NotificationId = notification.NotificationId;
  // .. populate
  return alert;
}
```

what do you think?